core/src/main/java/io/questdb/cairo/TxReader.java

```diff
@@ -0,0 +1,295 @@
+    }
+
+    public long getFixedRowCount() {
+        return fixedRowCount;
+    }
+
+    public long getMaxTimestamp() {
+        return maxTimestamp;
+    }
+
+    public long getMinTimestamp() {
+        return minTimestamp;
+    }
+
+    public long getPartitionSize(int i) {
+        return attachedPartitions.getQuick(i * LONGS_PER_TX_ATTACHED_PARTITION + PARTITION_SIZE_OFFSET);
+    }
+
+    public long getPartitionSizeByPartitionTimestamp(long ts) {
+        final int index = findAttachedPartitionIndex(getPartitionLo(ts));
+        if (index > -1) {
+            return attachedPartitions.getQuick(index + PARTITION_SIZE_OFFSET);
+        }
+        return -1;
+    }
+
+    public long getPartitionTableVersion() {
+        return partitionTableVersion;
+    }
+
+    public long getPartitionTimestamp(int i) {
+        return attachedPartitions.getQuick(i * LONGS_PER_TX_ATTACHED_PARTITION + PARTITION_TS_OFFSET);
+    }
+
+    public long getRowCount() {
+        return transientRowCount + fixedRowCount;
+    }
+
+    public long getStructureVersion() {
+        return structureVersion;
+    }
+
+    public long getTransientRowCount() {
+        return transientRowCount;
+    }
+
+    public long getTxn() {
+        return txn;
+    }
+
+    public long getTxEofOffset() {
+        return getTxMemSize(symbolsCount, attachedPartitions.size());
+    }
+
+    public void initPartitionBy(int partitionBy) {
+        this.timestampFloorMethod = partitionBy != PartitionBy.NONE ? getPartitionFloor(partitionBy) : null;
+        this.partitionBy = partitionBy;
+    }
+
+    public void open() {
+        assert this.roTxMem == null;
+        roTxMem = openTxnFile(ff, path, rootLen);
+        if (roTxMem instanceof ReadOnlyMemory) {
+            // In readonly mode Tx file has to call grow sometimes
+            this.readOnlyTxMem = (ReadOnlyMemory) roTxMem;
+        }
+    }
+
+    public void read() {
+        this.txn = roTxMem.getLong(TX_OFFSET_TXN);
+        this.transientRowCount = roTxMem.getLong(TX_OFFSET_TRANSIENT_ROW_COUNT);
+        this.fixedRowCount = roTxMem.getLong(TX_OFFSET_FIXED_ROW_COUNT);
+        this.minTimestamp = roTxMem.getLong(TX_OFFSET_MIN_TIMESTAMP);
+        this.maxTimestamp = roTxMem.getLong(TX_OFFSET_MAX_TIMESTAMP);
+        this.dataVersion = roTxMem.getLong(TX_OFFSET_DATA_VERSION);
+        this.structureVersion = roTxMem.getLong(TX_OFFSET_STRUCT_VERSION);
+        this.symbolsCount = roTxMem.getInt(TX_OFFSET_MAP_WRITER_COUNT);
+        this.partitionTableVersion = roTxMem.getLong(TableUtils.TX_OFFSET_PARTITION_TABLE_VERSION);
+        loadAttachedPartitions(this.maxTimestamp, this.transientRowCount);
+    }
+
+    public long readFixedRowCount() {
+        return roTxMem.getLong(TX_OFFSET_FIXED_ROW_COUNT);
+    }
+
+    public int readSymbolCount(int symbolIndex) {
+        return roTxMem.getInt(getSymbolWriterIndexOffset(symbolIndex));
+    }
+
+    public void readSymbolCounts(IntList symbolCountSnapshot) {
```

Comment:
can this be part of read()?

Reply:
Only TableReader need to read these counts, writer does not care about the values there so to save some memory / operations on TableWriter it's not in the read() but an additional call